src/components/MaterialUI/DataDisplay/TransferList.js

```diff
@@ -0,0 +1,174 @@
+		borderRadius: theme.spacing(0.5),
+		boxShadow: "none",
+
+		"&::-webkit-scrollbar": {
+			width: theme.spacing(1.5),
+		},
+		"&::-webkit-scrollbar-thumb": {
+			background: theme.palette.grey.borders,
+			border: `${theme.spacing(0.5)} white solid`,
+			backgroundClip: "padding-box",
+			borderRadius: theme.spacing(1.5),
+		},
+	},
+	item: {
+		fontSize: theme.spacing(1.3),
+		color: theme.palette.text.primary,
+		fontFamily: theme.typography.fontFamily,
+		padding: theme.spacing(0, 1.6),
+		cursor: "pointer",
+	},
+	selectedItem: {
+		backgroundColor: theme.palette.primary.light,
+	},
+
+	button: {
+		margin: theme.spacing(0.5, 0),
+		fontSize: theme.spacing(1.3),
+		lineHeight: theme.spacing(1.4),
+		padding: theme.spacing(0.7),
+		minWidth: theme.spacing(8),
+	},
+}));
+
+const formatState = (state, checked, from, to) => {
+	const fromSelectedItems = state[from].filter(({ id }) => checked.includes(id));
+	return {
+		[to]: [...state[to], ...fromSelectedItems].sort((a, b) => a.title.localeCompare(b.title)),
+		[from]: state[from].filter(({ id }) => !checked.includes(id)),
+	};
+};
+
+const CustomList = ({ items, checked, setChecked }) => {
+	const classes = useStyles();
+
+	const handleToggle = useCallback(
+		value =>
+			setChecked(state => {
+				const currentIndex = state.indexOf(value);
+				const newChecked = [...state];
+
+				if (currentIndex === -1) {
+					newChecked.push(value);
+				} else {
+					newChecked.splice(currentIndex, 1);
+				}
+				return newChecked;
+			}),
+		[setChecked],
+	);
+
+	return (
+		<List component="div" role="list">
+			{items.map(({ id, title, disabled }) => (
+				<ListItem
+					className={classNames(classes.item, checked.includes(id) && classes.selectedItem)}
+					key={id}
+					role="listitem"
+					onClick={!disabled ? () => handleToggle(id) : null}
+					disabled={disabled}
+				>
+					<ListItemText id={`transfer-list-item-${id}-label`} primary={title} />
+				</ListItem>
+			))}
+		</List>
+	);
+};
+
+const TransferList = ({
+	rightListData,
+	leftListData,
+	onChange,
+	addButtonTitle,
+	deleteButtonTitle,
+	customLeftComponent,
+}) => {
+	const classes = useStyles();
+	const [checked, setChecked] = useState([]);
+	const onTopButtonClick = () => {
+		onChange(formatState({ right: rightListData.data, left: leftListData.data }, checked, "left", "right"));
+		setChecked(checked.filter(check => !leftListData.data?.some(({ id }) => id === check)));
+	};
+	const onBottomButtonClick = () => {
+		onChange(formatState({ right: rightListData.data, left: leftListData.data }, checked, "right", "left"));
+		setChecked(checked.filter(check => !rightListData.data?.some(({ id }) => id === check)));
+	};
+
+	const rightSelected = rightListData.data?.some(({ id }) => checked.includes(id));
+	const leftSelected = leftListData.data?.some(({ id }) => checked.includes(id));
+
+	return (
+		<Grid container spacing={1} justify="center" alignItems="center" className={classes.root}>
+			<Grid item>
```

Comment:
Paper width should be responsive, I think Left and Right 5 cols, and for Buttons 2 cols